agent/config/conditional.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package config
+
+import (
+	"encoding/json"
+	"strconv"
+)
+
+// Conditional makes it possible to understand if a variable was set explicitly or relies on a default setting
+type Conditional int
+
+const (
+	_ Conditional = iota
```

Comment:
Does anything use this identifier?

Reply:
I wanted to avoid undesired/unexpected behavior when the iota is zero.